service/worker/scheduler/activities.go

```diff
@@ -65,12 +70,19 @@ type (
 	}
 )
 
+const (
+	eventStorageSize = 2 * 1024 * 1024
```

Comment:
This name is making a perfect sense. IMHO BlobSizeLimitError does not make any sence. You would not say it in English. If error needs to be produced once result size crossed eventStorageSize a function `shouldProduceError` or `aboveErrorSizeTreshold`. The fact that we used to use bad names does not mean that we should use bad names. Not sure if we can rename `BlobSizeLimitError` to something more appropriate though.

Reply:
You'd read it in English as "blob size limit, at the error level" (there is also a "warn" level).

I'm not saying any name has to read perfectly in English or that BlobSizeLimitError is a great name, I'm saying that this constant has exactly one use in non-test code: the default value for the dynamic config `dynamicconfig.BlobSizeLimitError`.
https://github.com/temporalio/temporal/pull/5369/files#diff-8af44785225f9c30fc88f4265638e6a1f89b2295c7cd609c8f74d5039645d0e3R91
It should be named based on what it is and what it does. Specifically, it does not define a limit on storage size of events, which the current name suggests. It's just a default value, and it's a default value for _blob_ size, which are eventually contained in events but the limit applies to the blob, not the whole event.

Reply:
Also, yes, the fact that we use bad names does mean that we should use bad names, because consistency is more important than the goodness of any specific name. If you want to fix a bad name, please do, but fix it everywhere, not just in one inconsistent corner.

Reply:
I like the second or third. (It's not just "reporting", it fails the workflow.) But I think the name of this constant should include `default`, because the actual error threshold can be changed using dynamic config, and it doesn't always equal this constant.